TCMSApp/src/client/app/defects/add-defect.controller.js

```diff
@@ -25,38 +27,81 @@
+                            var allowStateChange = false;
+                            vmDefectModal.bugName = '';
+                            vmDefectModal.whoFind = '';
+                            vmDefectModal.assignedTo = '';
+                            vmDefectModal.priority = 'Critical';
+                            vmDefectModal.dateOfDefectCreation = moment();
+                            vmDefectModal.description = '';
+                            vmDefectModal.chooseFile = '';
+                            vmDefectModal.stepsToReproduce = '';
+                            vmDefectModal.testCase = 'TODO';
+
+                            if ($stateParams.run) {
+                                vmDefectModal.run = $stateParams.run.info;
                             }
-                        });
-                    },
-                    controllerAs: 'vmDefectModal',
-                    backdrop: 'static'
-                });
+
+                            vmDefectModal.cancel = function () {
+                                allowStateChange = true;
+
+                                $uibModalInstance.dismiss('cancel');
+                                $state.go($stateParams.previousState);
+                            };
+
+                            /*function getNewDefect() {
+                             return dataservice.getNewDefect().then(function (data) {
+                             //var debug = data.getById("whoFind");
+                             return data;
+                             });
+                             }*/
+                            vmDefectModal.post = function () {
+                                allowStateChange = true;
+
+                                //var infoPromise = getNewDefect();
+                                var sample = {
+                                    name: vmDefectModal.bugName,
+                                    whoFind: vmDefectModal.whoFind,
+                                    assignedTo: vmDefectModal.assignedTo,
+                                    dateOfDefectCreation: vmDefectModal.dateOfDefectCreation, //TODO:check time
+                                    priority: vmDefectModal.priority.toLowerCase(),
+                                    chooseFile: vmDefectModal.chooseFile,
+                                    description: vmDefectModal.description,
+                                    stepsToReproduce: vmDefectModal.stepsToReproduce,
+                                    //testRunId:  vmDefectModal.testCase //TODO:
+                                };
+
+                                var defectsInfo = $resource(apiUrl.defects, {}, {});
+
+                                defectsInfo.save(sample);
+                                // updateDefects();
+                                $uibModalInstance.close();
+                                $state.go($stateParams.previousState, $stateParams, {reload: true, inherit: false,
```

Comment:
reload -  force transition even if the state or params have not changed, aka a reload of the same state.
inherit: false - it will not inherit url parameters from current url.
notify: true - If true will broadcast $stateChangeStart and $stateChangeSuccess events.